packages/math-poolsimulator/src/index.ts

```diff
@@ -75,9 +75,43 @@ export class AMM<TPoolPairData extends PoolPairData> {
         poolPairData,
         inGuessValue,
       );
-    const newInGuess = diffFromSpotPrice
-      .div(spotPriceDerivative)
-      .plus(inGuessValue);
+
+    let inGuessDelta = diffFromSpotPrice.div(spotPriceDerivative);
+    let newInGuess = inGuessValue.plus(inGuessDelta);
+
+    // Some pools are limited, so this iteration is to avoid a guess out of pool range.
+    for (let i = 0; i < 20; i++) {
+      if (this.math._checkIfInIsOnLimit(poolPairData, newInGuess)) break;
+      inGuessDelta = inGuessDelta.div(bnum(2));
+      newInGuess = inGuessValue.plus(inGuessDelta);
+    }
+
+    // This is the backtracking line search variation of the Newton-Raphson method
+    // that avoids guesses that don't significantly improve the result
+    const alpha = bnum(2);
+    const beta = bnum(8);
```

Comment:
is there anything special about these alpha/beta values? I assume they're not the same values from the pool params we're simulating, but rather only here for the numerical analysis part, right?

Reply:
I will add a comment about this, but the alpha and beta are from the algorithm. You have to set the alpha can be [2,infinity] and beta [1, infinity]